### ISSUE TITLE
miniz Zlib decompression

### ISSUE THREAD

**Issue by martinberlin** [martinberlin/udpx#13]
Hello Samuel @IoTPanic,
in this branch: https://github.com/martinberlin/udpx/tree/feature/miniz

Miniz reference: https://github.com/richgel999/miniz/releases

I'm trying to implement also gzip decompression just to compare and benchmark with Brotli, since it is supposed to compress about 10 faster than Brotli (Only theory and stats seen on the web)
But so far failing miserably, got a one file miniz.c to compile correctly although takes too much space and should be cleaned up for functions we won't use. File related maybe, since we only need to decompress a Buffer.

So basically last status is :
miniz compiled and working but still returns status MZ_DATA_ERROR

Trying it with: 
cat examples/72-on.bin.gz |nc -w1 -u ESP32_IP_ADDRESS 1234

Now one interesting thing is that a low level version of miniz [is already on ESP32 ROM](https://www.esp32.com/viewtopic.php?t=1076). The thing is that it includes only low level deinflate methods and I don't even know if I can compress something like that in Linux command line.
"the version of miniz in ROM is stripped back from the full library's features. As shown in the top of the header - the archive, zlib & zip APIs are not compiled in."

I thought maybe you can give me hand if you find time. Nothing urgent I'm just trying. 
If it works, I will take the time to search for a passing WebAssembly GZ compress library, one that looks interesting is:
https://github.com/drbh/wasm-flate 🔢

**Comment by IoTPanic** [comment]
Hm, sorry for not seeing this sooner. Are you able to access their stripped down library? I would think that would be compiled separately to your application and placed in a bootloader. And to add to your answer, can you access it in IDF if you can't in Arduino. 

The host machine has to be able to do the same compression so I'm sure either there is a way to compress already in the command line to this specification, or we can use ESPs implementation and compile it for a linux machine, than maybe compile to web assembly if that will work for you in the app.

**Comment by martinberlin** [comment]
Hi Samuel,
We need to do benchmarking / test and see if it's really worth it
I managed to get it working on ESP32 though adding miniz as an internal lib. Is working so far so I would not spend too much time on it until we prove it means something. 
Now for the ChromeApp/Android app I found this that looks fast:
https://github.com/drbh/wasm-flate

And I'm making a new Repository just with a web version that is simple to use and does not use eval() since the apps do not like evals due to security. Flate looks great since adds many compression/decompression algorithms, and for this case, we should use:

    flate.zlib_encode_raw(Buffer Uint8Array)

Thanks for your contribution. Let's try it first on Chrome App to see if it's really worth to use it.

**Comment by martinberlin** [comment]
App compression part is here:
https://github.com/martinberlin/udpx-app/tree/feature/9-zlib
Will release a new APK this time as a release in GitHub and assign it to you so is testeable.

If I detect no memory leaks, will merge this feature/miniz branch into master and at the same time clean up the BLE two APs mess to make it simpler. Then you can test it over and I will release next version over the weekend. Maybe @hputzek can test it too if he finds some time

**Comment by IoTPanic** [comment]
I never used zlib, and for some reason I thought it was another way to compress not a library which that is dumb, so I have spent the past few minutes pouring over text trying to figure out gzip vs zlib in your posts and tweets lmao. I feel dumb now. Thats pretty cool you got gzip working.

**Comment by martinberlin** [comment]
Thanks! Merged in master. Closing this after some more tests. Looks stable, I'm always initializing and deleting Buffers the way you tell me.

**Comment by martinberlin** [comment]
Implemented, tested and merged in master, however like I commented we are on the memory limit I think. Had to precisely adjust the Buffer size value otherwise won't work.
That's the thing with the libraries. I need to export a fucking 20 characters JSON and I use ArduinoJson that is huge. I need to take more care and stop using this, to stick with less memory hungry stuff. 
The ultimate goal is to have a stable Firmware that will work on differente environments and has space to breathe ;)
